### PR TITLE
Conan graph explain docs

### PR DESCRIPTION
**Pull request by danimtb**
Docs for https://github.com/conan-io/conan/pull/14694

### CODE REVIEW

File: reference/commands/graph.rst [hunk 1]
```diff
@@ -11,3 +11,4 @@ without needing to download the package binaries.
    conan graph info: Computes a dependency graph and displays information about it <graph/info>
    conan graph build-order: Computes the detailed sequence of packages that need to be built from source <graph/build_order>
    conan graph build-order-merge: Merge 2 existing build-order sequences into one <graph/build_order_merge>
+   conan graph expolain: Explain what is wrong with the dependency graph, showing missing binaries, closest alternatives, and why they do not matchç
```
Comment (memsharded) [inline reference/commands/graph.rst:14]:
```suggestion
   conan graph explain: Explain what is wrong with the dependency graph, showing missing binaries, closest alternatives, and why they do not match
```

File: reference/commands/graph/explain.rst [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      --lockfile-packages   Lock package-id and package-revision information
+      --lockfile-clean      Remove unused entries from the lockfile
+      --lockfile-overrides LOCKFILE_OVERRIDES
+                            Overwrite lockfile overrides
+      --check-updates       Check if there are recipe updates
+      --build-require       Whether the provided reference is a build-require
+      --missing [MISSING]   A pattern in the form 'pkg/version#revision:package_id#revision', e.g: zlib/1.2.13:* means all binaries for zlib/1.2.13. If revision is not specified, it is
+                            assumed latest one.
+
+The ``conan graph explain`` tries to give a more comprenhensive explanation for a package that might be missing with the configuration provided and show the differences between the expected binary package and the available one.
```
Comment (memsharded) [inline reference/commands/graph/explain.rst:84]:
```suggestion
The ``conan graph explain`` tries to give a more detailed explanation for a package that might be missing with the configuration provided and show the differences between the expected binary package and the available ones.
```

File: reference/commands/graph/explain.rst [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+It helps to understand what is missing from the package requested, wether it is different options, differnt settings or different dependencies.
```
Comment (memsharded) [inline reference/commands/graph/explain.rst:85]:
```suggestion
It helps to understand what is missing from the package requested, whether it is different options, different settings or different dependencies.
```

**Review by memsharded — COMMENTED** [review]
Would be good to update the occurrences of ``ERROR: Missing binary`` in the docs, there is at least one

**Comment by danimtb** [comment]
Done!

**Review by memsharded — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**